### PR TITLE
[TS] LPS-121719

### PR DESCRIPTION
**Pull request by ChrisKian**
https://issues.liferay.com/browse/LPS-121719

Forwarding to you from https://github.com/petershin/liferay-portal/pull/1263 as per @petershin's [request](https://github.com/petershin/liferay-portal/pull/1263#issuecomment-717365645).  Please let me know if you have any questions, thanks!

### CODE REVIEW

File: modules/apps/frontend-js/frontend-js-svg4everybody-web/src/main/resources/META-INF/resources/svg4everybody/svg4everybody.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
+!function(root, factory) {
```
Comment (jbalsas) [inline modules/apps/frontend-js/frontend-js-svg4everybody-web/src/main/resources/META-INF/resources/svg4everybody/svg4everybody.js:1]: Hey @ChrisKian, I think we need to be careful copy-pasting external code into DXP codebase without the proper attribution. Not being a lawyer, even for such permissive licenses as this one we should get in the habit of consulting with legal to see if this is the proper way to do it. I'd say this is at least missing the library license, but not sure how that should be brought in.

Hey @silverhook, I'm not sure this is what we want to do, but if it were, could you give us a quick overview of what would be the appropriate steps here?

**Comment by jbalsas** [comment]
Hey @ChrisKian, we're centralizing all the PRs the Frontend Infra team handles in the @liferay-frontend user. We also require that you adhere to some additional guidelines to improve the review process for all. Please take a look at the following documents:
- [Frontend Infrastructure Team code review process](https://github.com/liferay/liferay-frontend-guidelines/blob/master/dxp/code_review.md#frontend-infrastructure-team-code-review-process)
- [Pull Request Etiquette](https://github.com/liferay/liferay-frontend-guidelines/blob/master/general/pull_requests.md#pull-request-etiquette)

In summary:
- Send PRs to the @liferay-frontend user
- Add a descriptive **title**
- Add a proper b containing:
    - Context about the problem
    - Context about the solution
    - Things you considered 
- Add a **test plan**

We've found that adhering to these helps us provide better reviews in a more efficient way and appreciate your help with this! ❤️ 

PS: I think we could also squash all these commits into one. In some cases we have commits undoing previously done things.

**Comment by wincent** [comment]
To answer your question from [the original PR](https://github.com/petershin/liferay-portal/pull/1263):

> I'm not sure how to exclude the new JS file from source formatting. Is this something you can help with? Thanks!

You could add an `/* eslint-disable */` comment to the file, or add it to `modules/.eslintignore` file, but...

... what @jbalsas said is spot-on. svg4everybody [is public domain](https://github.com/jonathantneal/svg4everybody/blob/8bd711049ba3faefb8285ca70c63a67efc35636c/LICENSE.md), but even so, we

**Comment by wincent** [comment]
(Sorry about that, clicked "Close" by accident before I even finished typing that comment. I am going to edit it to finish it.)

**Comment by wincent** [comment]
And where I said `.eslintignore`, read also `.prettierignore` (🤦‍♂️ — sorry, it's late here).

**Comment by silverhook** [comment]
The first question we need to have answered is where this code came from and what license it is under.

As @wincent mentioned, public domain can be tricky, but not everything that lies around on the internet is in public domain – in fact the vast majority of it isn’t! What public domain means is simply that the copyright expired (or was explicitly waived, in the few jurisdictions that allow for that), which is quite unlikely.

For details please see the Inbound Licensing Policy on Grow.

**Comment by silverhook** [comment]
Oh, and if this is a new external component (it seems like it is), open up an _Inbound Licensing_ ticket in JIRA. There’s a template for that there, and I can take over then.

**Comment by wincent** [comment]
Closing this in favor of https://github.com/liferay-frontend/liferay-portal/pull/454

@ChrisKian: can you please take a look at that and verify that it still fixes the issue for you? I just verified that `svg4everybody()` builds and loads, but stopped there.

I squashed several of the commits (for example the ones that make changes and then revert them) into the following one:

---

#### LPS-121719 Use master version of svg4everybody.js

The project is pretty much unmaintained at this point: the last release was about 3 years ago, although the `master` branch has commits as recently as April this year (2020).

So, we take a snapshot of the source by grabbing the `LICENSE.md` and the `lib/svg4everybody.js` source file as of current HEAD of `master`, and basically concatenating them:

- https://raw.githubusercontent.com/jonathantneal/svg4everybody/8bd711049ba3faefb8285ca70c63a67efc35636c/LICENSE.md
- https://raw.githubusercontent.com/jonathantneal/svg4everybody/8bd711049ba3faefb8285ca70c63a67efc35636c/lib/svg4everybody.js

The project is simple, 0 dependencies, and 1 source file. We can ignore the Gruntfile based UMD build that the project uses and just compile this with `@liferay/npm-scripts`.

Changes made to the source:

-   Wrap whole thing in an IIFE and assign to `window.svg4everybody`.
-   Format source.
-   Fix lints:

        143:10  error  'svg4everybody' is defined but never used.  no-unused-vars
        153:6  error  'LEGACY_SUPPORT' is not defined.  no-undef
        187:13  error  'LEGACY_SUPPORT' is not defined.  no-undef
        245:9  error  'LEGACY_SUPPORT' is not defined.  no-undef
        302:9  error  Expected object keys to be in natural ascending order. 'id' should be before 'svg'.  sort-keys

    Note that we can just delete all the sections guarded by `LEGACY_SUPPORT` because we never used the legacy build.

Because the build is simpler now, we can strip out most of the `build.gradle` file.

**Test plan:**

-   In portal-ext.properties, set:

        cdn.dynamic.resources.enabled=false

    (This is the only way to inject svg4everybody, unless you are on IE, which you no longer can be on `master`).
-   Load portal and see script successfully retrieved from `/o/frontend-js-svg4everybody-web/index.js`.
-   See [transpiled source looks right](https://gist.github.com/wincent/5c61c838e0b8e3445e552c167cd9f2ef).
-   And that `svg4everybody()` exists in the console.
